### ISSUE TITLE
Exchange/standard format for taxonomies

### ISSUE THREAD

**Issue by brollb** [webgme/webgme-taxonomy#391]
It would be beneficial to have a standard representation for taxonomies and to update the generation logic (such as `JSONSchemaExporter`) to use this instead of the webgme core. This would be useful for:
- tagset implementation (tricky when terms go directly from `GMENode`s to JSON schema)
- decoupled data dashboard
- etc

Something like below may work:
```typescript
enum SelectionConstraint {
  Optional,
  Recommended,
  Required,
}
enum ReleaseState {
  Prerelease,
  Released,
  Deprecated,
}

type Guid = string;
type NodePath = string;
interface Vocabulary {
  id: Guid;
  source: NodePath;
  terms: Term[];
}

interface Term {
  id: Guid;
  name: string;
  selection: SelectionConstraint;
  fields: TermField[];
  state: ReleaseState;
  subTerms: Term[];  // I don't think we want this actually

  source: NodePath;
}

interface PrimitiveField {
  value?: any;
}

interface TermField {
  id: Guid;
  type: string;
  name: string;
  selection: SelectionConstraint;

  source: NodePath;
}

interface EnumField extends TermField {
  type = "enum";
  value? = Guid;
}

interface TextField extends TermField, PrimitiveField {
  type = "text";
  value? = string;
}

interface CompoundField extends TermField {
  type = "compound";
}
  
```

**Comment by brollb** [comment]
Here is my most recent pass at an exchange format for taxonomies (not tags). Feedback is appreciated!

```javascript
// First, this is how a few different field types are represented
{ Text: { value: 'defaultTextValue (optional)' } }
{ Uri: { value: 'defaultUriValue (optional)' } }

// Enum example
{
  Enum: {
    variants: [
      { Variant1: { ... } },
      { Variant2: { ... } },
    ]
  }
}

// Field type information is used in a field as shown below
{
  description: 'some description',
  content: { Text: { value: 'defaultTextValue' } },
  required: true,
}

// Field metadata noteable ignores the name as the name is specified in the term:
{
  id: 'someTermId',
  selection: 'required',
  readOnly: false,  // set to true for system terms since they are only used by users when reading content and cannot be created
  releaseState: 'deprecated',  // optional
  fields: {
    value: {  // "value" is the name of the field in this example
      id: 'fieldGuid',
      description: 'some description',
      content: { Uri: { value: 'defaultTextValue' } },
      required: true,
    }
  }
}

// Terms follow a similar pattern where the name is used as the key in the dictionary to prevent
// representing invalid terms (with name collisions)

// Term data is specified in a vocabulary, as expected:
{
  id: 'vocabularyGuid',
  releaseState: 'released',
  description: 'An example vocabulary',
  terms: {
    myCustomTerm: {  // "myCustomTerm" is the name of the term
      id: 'someTermId',
      selection: 'required',
      readOnly: false,
      fields: {
        value: {  // "value" is the name of the field in this example
          id: 'fieldGuid',
          description: 'some description',
          content: { Uri: { value: 'defaultTextValue' } },
          required: true,
        }
      }
    }
  }
}

// An example taxonomy represented in the exchange format is given as follows:
{
  version: '1.2.3',
  vocabularies: {
    myVocabularyName: {
      id: 'vocabularyGuid',
      releaseState: 'released',
      description: 'An example vocabulary',
      terms: {
        myCustomTerm: {  // "myCustomTerm" is the name of the term
          id: 'someTermId',
          selection: 'required',
          readOnly: false,
          fields: {
            value: {  // "value" is the name of the field in this example
              id: 'fieldGuid',
              description: 'some description',
              content: { Uri: { value: 'defaultTextValue' } },
              required: true,
            }
          }
        }
      }
    }
  }
}
```

**Comment by brollb** [comment]
Actually, enum variants should include their ID, name, and fields. Something like:
```javascript
{ id: "someId", name: "variantName", fields: {...} }
```

**Comment by brollb** [comment]
Closed by #472
